plugin/buycourses/src/buycourses.ajax.php

```diff
@@ -0,0 +1,226 @@
+        $pendingPayouts = $plugin->getPayouts(BuyCoursesPlugin::PAYOUT_STATUS_PENDING);
+        $canceledPayouts = $plugin->getPayouts(BuyCoursesPlugin::PAYOUT_STATUS_CANCELED);
+        $currency = $plugin->getSelectedCurrency();
+        
+        foreach ($completedPayouts as $completed) {
+            $stats['completed_count'] = count($completedPayouts);
+            $stats['completed_total_amount'] += $completed['comission'];
+            $stats['completed_total_amount'] = number_format($stats['completed_total_amount'], 2);
+        }
+        
+        foreach ($pendingPayouts as $pending) {
+            $stats['pending_count'] = count($pendingPayouts);
+            $stats['pending_total_amount'] += $pending['comission'];
+            $stats['pending_total_amount'] = number_format($stats['pending_total_amount'], 2);
+        }
+        
+        foreach ($canceledPayouts as $canceled) {
+            $stats['canceled_count'] = count($canceledPayouts);
+            $stats['canceled_total_amount'] += $canceled['comission'];
+            $stats['canceled_total_amount'] = number_format($stats['canceled_total_amount'], 2);
+        }
+        
+        
+        $html = '<div class="row">'
+        . '<p>'
+            . '<ul>'
+                . '<li>'. get_plugin_lang("PayoutsTotalCompleted", "BuyCoursesPlugin") .' <b>'. $stats['completed_count'] .'</b> - '. get_plugin_lang("TotalAmount", "BuyCoursesPlugin") .' <b>'. $stats['completed_total_amount'] .' '. $currency['iso_code'] . '</b></li>'
+                . '<li>'. get_plugin_lang("PayoutsTotalPending", "BuyCoursesPlugin") .' <b>'. $stats['pending_count'] .'</b> - '. get_plugin_lang("TotalAmount", "BuyCoursesPlugin") .' <b>'. $stats['pending_total_amount'] .' '. $currency['iso_code'] . '</b></li>'
+                . '<li>'. get_plugin_lang("PayoutsTotalCanceled", "BuyCoursesPlugin") .' <b>'. $stats['canceled_count'] .'</b> - '. get_plugin_lang("TotalAmount", "BuyCoursesPlugin") .' <b>'. $stats['canceled_total_amount'] .' '. $currency['iso_code'] . '</b></li>'
+            . '</ul>'
+        . '</p>';
+        $html .= '</div>';
+        echo $html;
+        
+        break;
+    
+    case 'processPayout':
+        if (api_is_anonymous()) {
+            break;
+        }
+        
+        $html = '';
+        $allPays = [];
+        $totalAccounts = 0;
+        $totalPayout = 0;
+        
+        $payouts = isset($_POST['payouts']) ? $_POST['payouts'] : '';
+        
+        if (!$payouts) {
+            
+            echo Display::return_message(get_plugin_lang("SelectOptionToProceed", "BuyCoursesPlugin"), 'error', false);
+            
+            break;
+        }
+        
+        foreach($payouts as $index => $id) {
+            $allPays[] = $plugin->getPayouts(BuyCoursesPlugin::PAYOUT_STATUS_PENDING, $id);
+        }
+        
+        foreach($allPays as $payout) {
+            $totalPayout += number_format($payout['comission'], 2);
```

Comment:
Please make sure this takes financial rounding rules into account. I think number_format(3.1487, 2) will transform it to 3.14, which is not correct. It should be 3.15. Generally speaking, rounding is left for the last step (the total sum) to avoid derivations during intermediate calculations